src/test/jtx/envconfig.h

```diff
@@ -0,0 +1,105 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_ENVCONFIG_H_INCLUDED
+#define RIPPLE_TEST_JTX_ENVCONFIG_H_INCLUDED
+
+#include <ripple/core/Config.h>
+
+namespace ripple {
+namespace test {
+
+/// @brief initializes a config object for use with jtx::Env
+///
+/// @param config the configuration object to be initialized
+extern
+void
+setupConfigForUnitTests (Config& config);
+
+namespace jtx {
+
+/// @brief creates and initializes a default
+/// configuration for jtx::Env
+///
+/// @return unique_ptr to Config instance
+inline
+std::unique_ptr<Config>
+envconfig()
+{
+    auto p = std::make_unique<Config>();
+    setupConfigForUnitTests(*p);
+    return p;
```

Comment:
Yes, I'd have it transfer ownership and then back out. All of these `mod` functions are meant to be used with `envconf` (if that wasn't the case, then passing the `unique_ptr` would be a bad interface). However, I don't see a scenario where they will be reused for anything other than `envconf`.

Pros:
* This would allow a function to create a completely new config if it desires
* The interface is clearer to me when modifications are returned rather than modifying a parameter (I dislike in/out parameters).

While I'll defend my POV, this is a minor issue. If you like the `in/out` parameter better it's your call.